visionipc/visionbuf_cl.cc

```diff
@@ -0,0 +1,93 @@
+#include "visionbuf.h"
+
+#include <stdio.h>
+#include <fcntl.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <sys/types.h>
+
+#define CL_USE_DEPRECATED_OPENCL_1_2_APIS
+#ifdef __APPLE__
+#include <OpenCL/cl.h>
+#else
+#include <CL/cl.h>
+#endif
+
+int offset = 0;
```

Comment:
this global isn't great, can easily see shadowing bugs from this

Reply:
also, make visionbuf a cpp class?

Reply:
The global was already there in old visionipc. If I turn it into an std::atomic<int> I think it should be safe. If you fork the process you get a different pid, so that should also be ok.